image-index.md

```diff
@@ -40,7 +41,7 @@ For the media type(s) that this document is compatible with, see the [matrix][ma
 
     Image indexes concerned with portability SHOULD use one of the above media types.
     Future versions of the spec MAY use a different mediatype (i.e. a new versioned format).
-    An encountered `mediaType` that is unknown SHOULD be safely ignored.
+    An encountered `mediaType` that is unspecified MUST be ignored.
```

Comment:
It's a bit ambiguous what "unspecified" means here -- does it mean that the implementation is not aware of the media-type or it's not defined in the OCI spec? If the former, then I'd prefer to use the word "unknown"  as before.

Reply:
```md
An encountered `mediaType` that is not [OCI-specified](media-types.md) or [compatible][matrix] MUST be ignored.
```

Reply:
> ```text
> An encountered `mediaType` that is not [OCI-specified](media-types.md) or [compatible][matrix] MUST be ignored.
> ```

Whoa -- why would you want to disallow people from having their own media-types? I think that the wording should be:

> An encountered `mediaType` that is unknown MUST be ignored.

Which would result in any out-of-spec mediaTypes being usable with any implementation (to the extent that they can be used without knowing what the media-type is).

The text you are proposing would **disallow implementations from defining their own out-of-spec media-types** (because any such implementation that didn't **ignore** their own media-type would not be following the OCI spec), which kind of defeats the purpose of them. The original text of this PR was just ambiguous, but that text is far less favourable. Not to mention I know of several organisations that are using their own media-types with OCI images, so such a change would be breaking them.

Reply:
The wording here is tricky.

The intention is that tools that are OCI compliant should not accidentally manipulate content that is being stored alongside it. They should all fail in the same way, which is to ignore them. 

The tools that store other artifacts are not going to be OCI-compliant. To do so, they'd have to implement a bunch of container-specific details. The idea is that these tools work alongside OCI in a way that doesn't break things.

The changes we are making to the specs are to enable the flexibility to experiment with storing this content without disrupting OCI. The intention is not to make OCI support arbitrary content because it's fundamentally a container standard.

Reply:
I understand that, and I agree. I don't agree that the current wording you've proposed does this -- in fact in my reading it actually disallows third-party `mediaType`s which is the opposite of what we both want.

I'm fine with @stevvooe's suggested wording:

> An encountered `mediaType` that is unknown to the implementation but unspecified by OCI MUST be ignored.

or the simpler variant I included above:

> An encountered `mediaType` that is unknown to the implementation MUST be ignored.

Which would result in `mediaType`s that are unknown to an implementation (meaning that it's an "other artifact" that the implementation doesn't know how to manage) then it won't touch it.